docs/api-guide/schemas.md

```diff
@@ -115,6 +115,67 @@ The `get_schema_view()` helper takes the following keyword arguments:
 * `renderer_classes`: May be used to pass the set of renderer classes that can
   be used to render the API root endpoint.
 
+### Handling `SerializerMethodField` returning a `dict`
+
+As Python is not a typed language, DRF cannot guess the return type of your method. By default, [`SerializerMethodField`][serializer-method-field] returned type in schema is a `string`. However, you may need to return another type.
+
+```python
+from django.contrib.auth.models import User
+from django.utils.timezone import now
+from rest_framework import serializers
+
+
+class UserSerializer(serializers.ModelSerializer):
+    days_and_hours_since_joined = serializers.SerializerMethodField()
+
+    class Meta:
+        model = User
+
+    def get_days_and_hours_since_joined(self, obj):
+        deltaT = now() - obj.date_joined
+        return {
+            'days_since_joined': deltaT.days,
+            'hours_since_joined': deltaT.seconds // 3600
+        }
+```
+
+In this example, `days_and_hours_since_joined` type in auto-generated schema type is `string`. It however should be an object containing 2 properties of type integer: `days_since_joined` and `hours_since_joined`.
+
+In order to workaround this, [nested serializers][nested-serializer] should be used instead.
```

Comment:
Well... OK, that's one way. 

I think the canonical answer here has to be to subclass AutoSchema and override `_map_serializer()` to return the right thing: 

```python
    def map_serializer(self, serializer):
        result = super().map_serializer(serializer)
        result['days_and_hours_since_joined'] = ... 
        return result
```

(I've called it `map_serializer()` here because we'll promote it to a _public_ stable method in the next release.) 

It's backwards to adjust your serializer for the sake of the schema.

Reply:
> I think the canonical answer here has to be to subclass AutoSchema and override `_map_serializer()` to return the right thing:
> 
> ```python
>     def map_serializer(self, serializer):
>         result = super().map_serializer(serializer)
>         result['days_and_hours_since_joined'] = ... 
>         return result
> ```

Problem with overriding at the view level is that you have to do the modification on each view using this serializer.
To go in your direction, ideal would be to have a method that can be overridden at the serializer level also.